### DISCUSSION TITLE
A solution to a common problem I have (goto log#)

### DISCUSSION THREAD

**Discussion by vesper8** [opcodesio/log-viewer#113] [Ideas]
I have a very common problem where I will occasionally get errors in my log, which will often contain thousands of entries such as debug/notice etc, and right next to the Error log will be some debug/notice logs that explain what caused the error. However I have no easy means of jumping to that point in the log history. 

I love that you add a number to each of the logs.. this is super convenient.. for example in my current log file I had an error which is number 131.. so using this I was able to paginate to this area and sure enough log 132 was a notice log that explained what caused 131

I guess it would be useful then to have a function that lets you jump to a numbered log.. similar to 'goto line' where it would jump to that log but also show the logs around this. I hope I'm making sense. : )

**Comment by arukompas** [comment]
@vesper8 hey! It's undocumented, and maybe not immediately clear, but there's already a way to jump to show individual log entries.

You can see logs have their numbers (indices) to the right side. They are clickable! This gives you a link to that individual log entry.

If you see what it's actually doing, is doing a **search** for `log-index:131`. That's your clue - just search for `log-index:XXXX` and you'll get that particular log.

Hope this helps :) 

**Otherwise**, if you know the log is as early as number 131, you can just order logs by earliest first, and then in just a few pages you'll find the log 131 & 132 & others following it.

Hope this helps :)

**Reply by vesper8** [reply]
@arukompas Thanks for letting me know they are clickable and provide a direct link to the log. This could be useful. However this doesn't solve my problem and my feature request above remains the same.

I would like the ability to "goto log" without isolating only that one log. I want to see the surrounding logs.. the surrounding logs are what are important. It's easy to find the log itself because it's often an error and it stands out.. plus the search makes it easy to find it. I really want to see the surrounding logs.. what happened before and after.

In my example with 131 then sure enough it's easy to sort and paginate a few times. But this is just an example. Most of my logs have 50k + log entries in it, what if my error log is 11238 and I want to see what 11239 and 11237 are? Then paginating becomes a nightmare.

Please consider adding the ability to go to an arbitrary log number but without isolating it. Just like a 'goto line' does in an IDE.

Many thanks!

**Reply by arukompas** [reply]
@vesper8 that totally makes sense! 👍 Thanks for the feedback 😄
